fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DataServlet.java

```diff
@@ -402,7 +399,7 @@ public void getData(
             Subject<PreResponse, PreResponse> queryResultsEmitter = PublishSubject.create();
 
             setupAsynchronousWorkflows(
-                    apiRequest,
+                    (DataApiRequestImpl) apiRequest,
```

Comment:
This cast indicates a problem. You can't assume a specific implementation of the interface.